backend/src/main.cpp

```diff
@@ -149,74 +150,51 @@ Napi::Boolean AddFrequency(const Napi::CallbackInfo& info)
         TRACK_LOG_WARNING("Could not add frequency, it already exists: {} {}", frequency, callsign);
         return Napi::Boolean::New(info.Env(), false);
     }
-    mClient->SetRx(frequency, false);
-    mClient->SetRadioGainAll(UserSession::currentRadioGain);
 
-    MainThreadShared::mApiServer->handleAFVEventForWebsocket(
-        sdk::types::Event::kFrequencyStateUpdate, {}, {});
+    RadioState newState;
+
+    newState.frequency = frequency;
+    newState.rx = false;
+    newState.tx = false;
+    newState.xc = false;
+    newState.headset = true;
+    newState.xca = false;
 
-    return Napi::Boolean::New(info.Env(), true);
+    auto result = RadioHelper::SetRadioState(MainThreadShared::mApiServer.get(), newState);
+    return Napi::Boolean::New(info.Env(), result);
 }
 
 void RemoveFrequency(const Napi::CallbackInfo& info)
 {
-    int frequency = info[0].As<Napi::Number>().Int32Value();
-    mClient->SetRx(frequency, false);
-    mClient->SetTx(frequency, false);
-    mClient->SetXc(frequency, false);
-    mClient->SetCrossCoupleAcross(frequency, false);
-    mClient->RemoveFrequency(frequency);
-    MainThreadShared::mApiServer->handleAFVEventForWebsocket(
-        sdk::types::Event::kFrequencyStateUpdate, {}, {});
+    RadioState newState;
+
+    newState.frequency = info[0].As<Napi::Number>().Int32Value();
+    newState.rx = false;
+    newState.tx = false;
+    newState.xc = false;
+    newState.headset = false;
+    newState.xca = false;
+
+    RadioHelper::SetRadioState(MainThreadShared::mApiServer.get(), newState);
```

Comment:
Use the new shared method for setting state.

Reply:
Same here, no raw pointers please